AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_LightAdded.py

```diff
@@ -97,74 +182,299 @@ def AtomEditorComponents_Light_AddedToEntity():
+            # 2. Remove existing light component from entity.
+            if light_entity.has_component(AtomComponentProperties.light()):
+                light_entity.remove_component(AtomComponentProperties.light())
+                Report.critical_result(Tests.light_component_removal,
+                                       not light_entity.has_component(AtomComponentProperties.light()))
 
-        # 4. REDO the entity creation and component addition.
-        # -> REDO entity creation.
-        general.redo()
-        # -> REDO selecting entity.
-        general.redo()
-        # -> REDO naming entity.
-        general.redo()
-        # -> REDO component addition.
-        general.redo()
-        general.idle_wait_frames(1)
-        Report.result(Tests.creation_redo, light_entity.exists())
+            # 3. Add Light component to the Light entity.
+            light_component = light_entity.add_component(AtomComponentProperties.light())
+            if iterations == 0:
+                Report.critical_result(Tests.light_component,
+                                       light_entity.has_component(AtomComponentProperties.light()))
 
-        # 5. Cycle through all light types.
-        for light_type in LIGHT_TYPES.keys():
+                # 4. UNDO the entity creation and component addition.
+                # -> UNDO component addition.
+                general.undo()
+                # -> UNDO naming entity.
+                general.undo()
+                # -> UNDO selecting entity.
+                general.undo()
+                # -> UNDO entity creation.
+                general.undo()
+                general.idle_wait_frames(1)
+                Report.result(Tests.creation_undo, not light_entity.exists())
+
+                # 5. REDO the entity creation and component addition.
+                # -> REDO entity creation.
+                general.redo()
+                # -> REDO selecting entity.
+                general.redo()
+                # -> REDO naming entity.
+                general.redo()
+                # -> REDO component addition.
+                general.redo()
+                general.idle_wait_frames(1)
+                Report.result(Tests.creation_redo, light_entity.exists())
+
+                # Update loop iterations to prevent additional running of these tests
+                iterations += 1
+
+            # 6. Set light type.
             light_component.set_component_property_value(
                 AtomComponentProperties.light('Light type'), LIGHT_TYPES[light_type])
             current_light_type = light_component.get_component_property_value(
                 AtomComponentProperties.light('Light type'))
             test_light_type = (
-                f"Light component has {light_type} type set",
-                f"Light component failed to set {light_type} type")
+                f"Set light type: {light_type.upper()}",
+                f"P0: Light component failed to set {light_type} type")
```

Comment:
why only success gets .upper()

Reply:
Updated so that the failure also gets .upper()